base/abstractarray.jl

```diff
@@ -902,7 +902,7 @@ end
 function hvcat_fill(a, xs)
     k = 1
     nr, nc = size(a,1), size(a,2)
-    for i=1:nr
+    for i=1:nr #Fixme iter
```

Comment:
This function appears to be called only on `Array`s, so you can simply restrict the signature and keep the existing code. Likewise, `xs` can be marked as `::Number`.